### ISSUE TITLE
Postgis Plugin: ERROR:  could not resize shared memory segment XXX to YYY bytes: No space left on device,

### ISSUE THREAD

**Issue by gakowalski** [pot-gov-pl/openstreetmap-tile-server#3]
I've encountered strange error which says that `No space left on device` while there is plenty of free space and also it says in one line `renderd[126]: ERROR: failed to render TILE ajt 6 32-39 16-23,` and in the next line `renderd[126]: DEBUG: DONE TILE ajt 6 32-39 16-23 in 4.182 seconds,`. I don't know if it's important but it's worth reporting.

> renderd[126]: ERROR: failed to render TILE ajt 6 32-39 16-23,
> renderd[126]:    reason: Postgis Plugin: ERROR:  could not resize shared memory segment "/PostgreSQL.609248538" to 50438144 bytes: No space left on device,
> in executeQuery Full sql was: 'SELECT ST_AsBinary("way") AS geom,"feature","way_pixels" FROM (SELECT,
>     way, way_pixels,,
>     COALESCE(wetland, landuse, "natural") AS feature,
>   FROM (SELECT,
>       way,,
>       ('landuse_' || (CASE WHEN landuse IN ('forest', 'farmland', 'residential', 'commercial', 'retail', 'industrial',,
>                                             'meadow', 'vineyard', 'orchard') THEN landuse ELSE NULL END)) AS landuse,,
>       ('natural_' || (CASE WHEN "natural" IN ('wood', 'sand', 'scree', 'shingle', 'bare_rock', 'heath', 'grassland', 'scrub') THEN "natural" ELSE NULL END)) AS "natural",,
>       ('wetland_' || (CASE WHEN "natural" IN ('wetland', 'mud') THEN (CASE WHEN "natural" IN ('mud') THEN "natural" ELSE tags->'wetland' END) ELSE NULL END)) AS wetland,,
>       way_area/NULLIF(POW(8.73566e+06*0.001*0.28,2),0) AS way_pixels,,
>       way_area,
>     FROM planet_osm_polygon,
>     WHERE (landuse IN ('forest', 'farmland', 'residential', 'commercial', 'retail', 'industrial', 'meadow', 'vineyard', 'orchard'),
>       OR "natural" IN ('wood', 'wetland', 'mud', 'sand', 'scree', 'shingle', 'bare_rock', 'heath', 'grassland', 'scrub')),
>       AND way_area > 0.01*2445.98::real*2445.98::real,
>       AND building IS NULL,
>   ) AS features,
>   ORDER BY way_area DESC, feature,
> ) AS landcover_low_zoom WHERE "way" && ST_SetSRID('BOX3D(-313086.0678562508 4696291.017843749,5322463.153556248 10331840.23925625)'::box3d, 3857)',
> renderd[126]: DEBUG: DONE TILE ajt 6 32-39 16-23 in 4.182 seconds,

**Comment by gakowalski** [comment]
The Internet suggests that this might be related to [shared memory settings for docker container](https://meta.discourse.org/t/pg-throws-could-not-resize-shared-memory-segment-error/84744/5). Probably there is a need to run container with switch `--shm-size` set to higher value than default  `64m`.

Worth checking.

**Comment by gakowalski** [comment]
I've doubled default 64m to `--shm-size="128m"` but errors were still present.

Then I doubled it again to `--shm-size="256m"` and errors were gone but memory and processor usage went abnormally high.

Then I lowered it to `--shm-size="192m"`. No errors, memory and processor on normal levels.

I guess, it's important to find some proper shared memory limit. Too low and too high values cause errors.
